src/Access/SettingsConstraints.cpp

```diff
@@ -107,6 +128,34 @@ void SettingsConstraints::clamp(const Settings & current_settings, SettingsChang
         });
 }
 
+template <class T>
+bool getNewValueToCheck(const T & current_settings, SettingChange & change, Field & new_value, bool throw_on_failure)
+{
+    Field current_value;
+    bool has_current_value = current_settings.tryGet(change.name, current_value);
+
+    /// Setting isn't checked if value has not changed.
+    if (has_current_value && change.value == current_value)
+        return false;
+
+    if (throw_on_failure)
+        new_value = T::castValueUtil(change.name, change.value);
+
+    try
+    {
+        new_value = T::castValueUtil(change.name, change.value);
+    }
+    catch (...)
+    {
+        return false;
+    }
```

Comment:
this should be under else, otherwise you're casting twice

Reply:
Thanks!